pkg/runctl/run_manager.go

```diff
@@ -643,7 +655,14 @@ func (c *runManager) addTektonTaskRunParamsForLoggingElasticsearch(
 func (c *runManager) GetRun(pipelineRun k8s.PipelineRun) (runi.Run, error) {
 	namespace := pipelineRun.GetRunNamespace()
 	run, err := c.factory.TektonV1alpha1().TaskRuns(namespace).Get(tektonTaskRunName, metav1.GetOptions{})
-	return NewRun(run), err
+	if k8serrors.IsResourceExpired(err) || k8serrors.IsGone(err) {
```

Comment:
do we know more cases where retry will not help?

Reply:
Changed this to whitelist, see 18d017ec8d055cd8fa7c4388126fe3916c97fe7c